core/core-impl/src/main/java/ai/timefold/solver/core/impl/solver/DefaultSolver.java

```diff
@@ -302,16 +330,20 @@ public void solvingEnded(SolverScope<Solution_> solverScope) {
     }
 
     public void outerSolvingEnded(SolverScope<Solution_> solverScope) {
-        // Must be kept open for doProblemFactChange
-        solverScope.getScoreDirector().close();
         logger.info("Solving ended: time spent ({}), best score ({}), score calculation speed ({}/sec), "
+                + "entity count ({}), variable count ({}), maximum value range size ({}), "
                 + "phase total ({}), environment mode ({}), move thread count ({}).",
                 solverScope.getTimeMillisSpent(),
                 solverScope.getBestScore(),
                 solverScope.getScoreCalculationSpeed(),
+                solverScope.getProblemStatistics().entityCount(),
+                solverScope.getProblemStatistics().variableCount(),
+                solverScope.getProblemStatistics().maximumValueRangeSize(),
                 phaseList.size(),
                 environmentMode.name(),
                 moveThreadCountDescription);
```

Comment:
I'm thinking that maybe we introduce a new logging line for problem statistics?
It seems to me that the counts and sizes are a different kind of information than the env mode, move thread count etc. - one is problem-specific, one is solver configuration.

Not entirely sure about it, but the current approach seems to me like information overload. Too much information at the same place.